### PR TITLE
azure - unify autotag for event\resource

### PR DESCRIPTION
**Pull request by logachev**
- Add new property `default-claim` which allows pick either `upn` or `name` claim as a default
- Use same approach for event-based and regular run usages
- Update tests

closes https://github.com/cloud-custodian/cloud-custodian/issues/3702
closes https://github.com/cloud-custodian/cloud-custodian/issues/4407
closes https://github.com/cloud-custodian/cloud-custodian/issues/5297

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/actions/tagging.py [hunk 1]
```diff
@@ -281,21 +287,27 @@ def _get_tag_value_from_event(self, event):
...
 
-        # Other types and main fallback (e.g. User, Office 365 Groups, and Security Groups)
-        if not user and self.upn_jmes_path.search(event):
-            user = self.upn_jmes_path.search(event)
+        if not user:
+            known_claims = {'upn': self.upn_jmes_path.search(event),
+                            'name': self.name_jmes_path.search(event)}
+            if known_claims[self.default_claim]:
+                user = known_claims[self.default_claim]
+            elif self.default_claim == 'upn' and known_claims['name']:
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/actions/tagging.py]: Is it not sufficient to just set the user = known_claims['name'] or user = known_claims['upn'] if values exist in `known_claims`?

Does it matter what the default_claim is?

File: tools/c7n_azure/c7n_azure/actions/tagging.py [hunk 2]
```diff
@@ -281,21 +287,27 @@ def _get_tag_value_from_event(self, event):
         elif StringUtils.equal(principal_type, 'ServicePrincipal'):
             user = self.sp_jmes_path.search(event)
 
-        # Other types and main fallback (e.g. User, Office 365 Groups, and Security Groups)
-        if not user and self.upn_jmes_path.search(event):
-            user = self.upn_jmes_path.search(event)
+        if not user:
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/actions/tagging.py]: We should add more unit tests that walk through this fallback logic, where if one default claim is missing we try the other.

Comment (logachev) [reply tools/c7n_azure/c7n_azure/actions/tagging.py]: Good point, thanks!

File: tools/c7n_azure/c7n_azure/actions/tagging.py (hunk 1 above)
Comment (logachev) [reply tools/c7n_azure/c7n_azure/actions/tagging.py]: They can exist at the same time.. I'm not 100% sure in what case they might be different, but I'd expect it to be the case.

**Pull request merged**